pkg/k8splan/watcher.go

```diff
@@ -153,64 +192,111 @@ func (w *watcher) start(ctx context.Context) {
 				}
 			}
 
+			// Check to see if we've exceeded our failure count threshold
+			var maxFailureThreshold int
+			if rawMaxFailureThreshold, ok := secret.Data[maxFailuresKey]; ok {
+				// max failure threshold is defined. parse and compare
+				maxFailureThreshold, err = strconv.Atoi(string(rawMaxFailureThreshold))
+				if err != nil {
+					maxFailureThreshold = int(^uint(0) >> 1) // max int
+				}
+			} else {
+				maxFailureThreshold = int(^uint(0) >> 1) // max int
+			}
+
+			logrus.Debugf("[K8s] Parsed max failure threshold value of %d", maxFailureThreshold)
+			wasFailedPlan := false
+			if rawFailureCount, ok := secret.Data[failureCountKey]; ok {
+				failureCount, err := strconv.Atoi(string(rawFailureCount))
+				if err == nil {
```

Comment:
Can we reverse this logic and check for the error and just log it if it exists? We an move the logic in this block out of this indentation.

Reply:
I didn't want to have the rest of this code hit in the event that the conversion was unsuccessful as we don't want to cause false positives of failure detection. It should basically never be that way unless someone manually manipulates the count value in the secret or corruption occurs.